inbm/dispatcher-agent/dispatcher/source/source_command.py

```diff
@@ -120,7 +120,7 @@ def _handle_app_source_command(
             logger.info(f"Optional GPG key parameters not present in manifest")
```

Comment:
**Suggestion:** Refactor the repeated logic for extracting 'gpg_key_name' and 'gpg_key_uri' into a separate function to avoid code duplication and improve maintainability. [maintainability]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
            def extract_gpg_keys(parsed_head, action):
                gpg_keys = {'uri': None, 'keyname': None}
                try:
                    gpg_keys = parsed_head.get_children(f"applicationSource/{action}/gpg")
                except XmlException:
                    logger.info(f"Optional GPG key parameters not present in manifest for {action}")
                return gpg_keys['uri'], gpg_keys['keyname']
    
            gpg_key_uri, gpg_key_name = extract_gpg_keys(parsed_head, 'add')
```

Reply:
making that call is a reuse.  It's down to a one line call to get_children which does all the work.